### PR TITLE
Added caveats for TimeToBeReceived

### PR DESCRIPTION
**Pull request by janovesk**
Caveats about the different transport implementations and how that affects the usefulness of this feature on the different transports.

### CODE REVIEW

File: nservicebus/messaging/discard-old-messages.md [hunk 1]
```diff
@@ -29,3 +29,17 @@ To discard a message when a specific time interval has elapsed:
 
 <!-- import DiscardingOldMessagesWithCode -->
 
+## Caveats
+TimeToBeReceived relies on underlying functionality in the transport infrastructure to discard expired messages. This features usefulness is highly affected by the actual implementation in the different transports.
```
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages.md]: `feature's`?

File: nservicebus/messaging/discard-old-messages.md [hunk 2]
```diff
@@ -29,3 +29,17 @@ To discard a message when a specific time interval has elapsed:
...
+
+#### MSMQ transport
```
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages.md]: Why `####` and not `###`?

File: nservicebus/messaging/discard-old-messages.md [hunk 3]
```diff
@@ -29,3 +29,17 @@ To discard a message when a specific time interval has elapsed:
...
+MSMQ continuously checks the TimeToBeReceived attribute of all queued messages. As soon as the message has expired, it is removed from the queue, and disk space reclaimed. Note that underlying issues in the MSMQ infrastructure makes it impossible for us to support setting TimeToBeReceived for messages sent from a transactional endpoint.
```
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages.md]: `make it impossible`

File: nservicebus/messaging/discard-old-messages.md [hunk 4]
```diff
@@ -29,3 +29,17 @@ To discard a message when a specific time interval has elapsed:
...
+
+#### RabbitMQ transport
+RabbitMQ also continuously checks the TimeToBeReceived attribute, but only for the top message on each queue. Expired messages are not removed from the queue, and their disk space is not reclaimed until they reach the front of the queue. Using TimeToBeReceived as a disk saving measure on RabbitMQ is not a great choice for queues with long-lived messages like audit and forward. The TimeToBeReceived of other messages in front of a message in a queue will affect when the message is actually removed.
```
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages.md]: I think queues have first or last message, but top makes more sense for stacks.

File: nservicebus/messaging/discard-old-messages.md (hunk 3 above)
Comment (danielmarbach) [inline nservicebus/messaging/discard-old-messages.md]: I'd add more explanation what the underlying issues are.

File: nservicebus/messaging/discard-old-messages.md (hunk 4 above)
Comment (SzymonPobiega) [inline nservicebus/messaging/discard-old-messages.md]: Hmm if I am overriding TTBR for audits and doing it consistently for each message, does it mean that the disk space will be reclaimed? Or do I actually need to actively receive a message in the front for rabbit to check the TTBR and delete that message?

Comment (janovesk) [inline nservicebus/messaging/discard-old-messages.md]: You don't have to receive with RabbitMQ, but the message has to be the first message.

**Pull request merged**